app/javascript/components/EditAccount.jsx

```diff
@@ -0,0 +1,266 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { Upload, message, Form, Icon, Select, Input, Button, Slider, Switch, DatePicker, Row, Col } from 'antd';
+import "antd/dist/antd.css";
+import moment from 'moment';
+import APIRoutes from 'helpers/api_routes';
+import Utils from 'helpers/utils';
+import ActiveStorageProvider from "react-activestorage-provider";
+import '../../assets/stylesheets/application.css';
+
+
+class EditAccount extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      user: this.props.user,
+      email: this.props.email,
+      new_password: this.props.current_password,
+      confirm_password: null,
+      disabled: false //to prevent multiple form submissions
+    };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleEdit = this.handleEdit.bind(this);
+    this.handleDestroy = this.handleDestroy.bind(this);
+  }
+
+  /* takes in two arrays (our array of field names and our array of values)
+   * and combines the corresponding field and value as a key value pair to be
+   * returned as a dictionary that will be sent in our JSON requests.
+   */
+  convertToDict() {
+    const user = this.state.user;
+    const keys = ["name", "address", "email", "phone", "password"];
+    const values = [user.name, user.address, this.state.email, user.phone, this.state.new_password];
+    let result = keys.reduce((obj, k, i) => ({...obj, [k]: values[i] }), {})
+    return result
+  }
+
+  //updates our values
+  handleChange = (e) => {
+    this.state.user[e.target.name] = e.target.value;
+  }
+
+  handleEmailChange = (e) => {
+    this.setState({ email: e.target.value });
+  }
+
+  //api destroy
+  handleDestroy() {
+    let id = this.props.user.id;
+    let type = this.props.type;
+    var request = null;
+    if (this.props.type === "landlord") {
```

Comment:
Cleaned up this part of the code because we no longer need to take into account of property/tenant.